src/OrleansSQLUtils/Storage/RelationalOrleansQueries.cs

```diff
@@ -833,13 +833,26 @@ private static ReminderEntry CreateReminderEntry(IDataRecord record)
 
         private static IDbDataParameter CreateBeginHashParameter(IDbCommand command, uint beginHash)
         {            
-            return command.CreateParameter(ParameterDirection.Input, "beginHash", (int)beginHash);
+            return CreateHashRangeParameter(command, "beginHash", beginHash);
         }
 
 
         private static IDbDataParameter CreateEndHashParameter(IDbCommand command, uint endHash)
         {            
-            return command.CreateParameter(ParameterDirection.Input, "endHash", (int)endHash);
+            return CreateHashRangeParameter(command, "endHash", endHash);
+        }
+
+
+        private static IDbDataParameter CreateHashRangeParameter(IDbCommand command, string paramName, uint hash)
+        {
+            int intRange;
+            unchecked
+            {
+                const uint delta = (uint)int.MinValue;
```

Comment:
can you add a detailed comment here, explaining the original problem and the solution.